internal/logstreamer/syncbuf.go

```diff
@@ -0,0 +1,73 @@
+package logstreamer
+
+import (
+	"bytes"
+	"sync"
+)
+
+// SyncBuf is basically a buffer with locks and
+// updates from broker subscriber
+type SyncBuf struct {
+	mu  sync.RWMutex
+	buf *bytes.Buffer
+}
+
+func NewSyncBuf(sub *BrokerSub) *SyncBuf {
+	sb := &SyncBuf{
+		buf: &bytes.Buffer{},
+	}
+
+	go func() {
+		for pack := range sub.load {
+			sb.Write(pack)
+		}
```

Comment:
I may be missing something, but it's not very clear why different subscribers write to the same buffer?

Reply:
I think `sub` is a single subscriber and this is a range loop over its channel of updates. Feels wrong for the "buffer" to handle reading the subscription, probably something outside the buffer should read the subscription and write the updates to the buffer